project4/3700crawler

```diff
@@ -81,21 +94,56 @@ class WebCrawler:
             except:
                 break
         # Close the socket
-        sock.close()
+        sock.close() 
+        # Return the data
         return data
 
+    def parse_response(self, data):
+        """Returns a dict of the headers and HTML"""
+        response = {}
+        data = data.split(CRLF + CRLF)
+        # Get the headers
+        headers = data[0]
+
+        # Add the status code and message
+        status = re.search('(HTTP/1.1)\W(\d{3})\W(.+)', headers)
+        response['Status Code'] = status.group(2) if status else None
+        response['Status Message'] = status.group(3) if status else None
+
+        # Set the CSRF token, session ID, and cookie
+        self.set_cookie(headers)
```

Comment:
I don't like this approach, but I'll approve it and refactor when I rebase my commit atop yours.

Reply:
Yeah, it doesn't abort when necessary. Let me try refactoring it first based on what I think you'd want to do instead.

Reply:
Nvm, I'll save the refactoring for another PR.